pymongo/pool.py

```diff
@@ -894,7 +895,7 @@ def __init__(self, address, options, handshake=True):
         # Can override for testing: 0 to always check, None to never check.
         self._check_interval_seconds = 1
 
-        self.sockets = set()
+        self.sockets = collections.deque()
```

Comment:
Can you add a comment that explains `sockets` is ordered with the oldest sockets on the right?

Reply:
Sure. I've added the comment